pkg/provision/engine.go

```diff
@@ -88,6 +92,7 @@ func New(opts EngineOps) *Engine {
 		statser:        opts.Statser,
 		zbusCl:         opts.ZbusCl,
 		janitor:        opts.Janitor,
+		memStats:       opts.MemStats,
```

Comment:
I will probably only keep the `Total` memory instead of the entire struct (while it's not a problem) . But what we definitely need is to not allocate the full memory. ZOS needs at least 2GB of memory for it's services and rootfs.

So my suggestion is, engine can keep a single value of allowed memory which is `total - const` where the const is 2G